concent_api/api_testing_common.py

```diff
@@ -120,17 +132,17 @@ def validate_content_type(actual_content_type, expected_content_type):
 
 
 def api_request(
-    host,
-    endpoint,
-    private_key,
-    public_key,
-    data=None,
-    headers=None,
-    expected_status=None,
-    expected_message_type=None,
-    expected_content_type=None
-):
-    def _prepare_data(data):
+    host: str,
+    endpoint: str,
+    private_key: bytes,
+    public_key: bytes,
+    data: Optional[Union[bytes, str]]=None,
+    headers: Optional[dict]=None,
+    expected_status: Optional[int]=None,
+    expected_message_type: Optional[Message]=None,
+    expected_content_type: Optional[str]=None
+) -> Union[None, Message]:
+    def _prepare_data(data: Optional[Union[bytes, str]]=None) -> Union[str, bytes]:
```

Comment:
> I would do it different. In api_request signature data can be None, but in real it has no sense and we never use api_request without data. Without golem message such request will not pass validations.

It would require changing the signature of `api_request` which could be a candidate for a separate request, because it changes functionality (`None` was there already) and would like to avoid doing this in this pull request. But I think it should be still possible to send this kind of request to Concent. For example, we might want to check what will happen if `receive` is being hit without authentication message.